src/state.rs

```diff
@@ -63,14 +65,48 @@ impl State {
         let next_state = State {
             player: self.player,
             board: self.board.move_piece(from_pos, to_pos),
+            en_passant: None,
         };
         !next_state.in_check()
     }
 
+    fn en_passant_pos(&self, from: Pos, to: Pos) -> Option<Pos> {
+        match self.board.piece_at(from) {
```

Comment:
I might do this with `self.board.piece_at.and_then(...)` since it feels more idiomatic than a `match` statement.

Reply:
the awkward bit about this is that i'm throwing out all `Some` values that are not pawns along with `None`, so even with `and_then` i still need a match statement, i think